nectar/src/command/trade/event_loop.rs

```diff
@@ -21,13 +21,10 @@ use comit::{
     orderpool::Match,
     Position, Role,
 };
-use futures::{
-    channel::mpsc::{Receiver, Sender},
-    FutureExt, StreamExt, TryFutureExt,
-};
+use futures::{channel::mpsc::Receiver, FutureExt, StreamExt, TryFutureExt};
 use std::sync::Arc;
 
-pub struct EventLoop {
+pub(super) struct EventLoop {
```

Comment:
Why do you need `pub(super)`? You could just make the struct `pub`, the module private and not re-export it from the parent module. That should achieve the same thing without special visibility modifiers :)

Reply:
Hum, I'll double check what I was doing but rustc wasn't happy as `TaskExecutor` is part of the signature but it is private to `trade` module.

Reply:
As discussed, we cannot go around this one.